src/openrct2-ui/windows/Options.cpp

```diff
@@ -1164,15 +1164,16 @@ static void window_options_mousedown(rct_window* w, rct_widgetindex widgetIndex,
                     audio_populate_devices();
 
                     // populate the list with the sound devices
-                    for (size_t i = 0; static_cast<int32_t>(i) < gAudioDeviceCount; i++)
+                    for (int32_t i = 0; i < audio_get_device_count(); ++i)
                     {
+                        auto& deviceName = audio_get_device_name(i);
                         gDropdownItemsFormat[i] = STR_OPTIONS_DROPDOWN_ITEM;
-                        gDropdownItemsArgs[i] = reinterpret_cast<uintptr_t>(gAudioDevices[i].name);
+                        gDropdownItemsArgs[i] = reinterpret_cast<uintptr_t>(deviceName.c_str());
```

Comment:
Is `gDropdownItemsArgs` an array of strings or of `char*´? Because if it's the latter, this would save the pointer of a soon to be unscoped variable (deviceName), so we'd have to `strdup`

Reply:
It returns std::string as reference and the audio devices are populated at the very beginning so this pointer will be alive until the application closes, this should be fine and has been like that before. I know it aint pretty but its been like this before.